File: app/src/main/java/org/cerion/stockcharts/watchlist/RecyclerViewAdapter.java
package org.cerion.stockcharts.watchlist;


import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.cerion.stockcharts.charts.ChartsActivity;
import org.cerion.stockcharts.databinding.ListItemWatchBinding;

import java.util.List;

class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ListItemViewHolder> {

    private List<WatchItemViewModel> items;

    RecyclerViewAdapter(List<WatchItemViewModel> items) {
        this.items = items;
    }

    @Override
    public ListItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        ListItemWatchBinding binding = ListItemWatchBinding.inflate(layoutInflater, parent, false);
        return new ListItemViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(ListItemViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    private static int defaultColor = -1;
    public final class ListItemViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        ListItemWatchBinding binding;

        public ListItemViewHolder(ListItemWatchBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
            this.binding.weekRange.setEnabled(false);
            this.binding.yearRange.setEnabled(false);

            binding.getRoot().setOnClickListener(this);
        }

        public void bind(WatchItemViewModel item) {
            binding.setViewModel(item);
            binding.executePendingBindings();
            item.load();
        }

        @Override
        public void onClick(View v) {
            Intent intent = ChartsActivity.newIntent(v.getContext(), binding.getViewModel().getSymbol(), binding.getViewModel().getChart());
            v.getContext().startActivity(intent);
        }
    }
}

File: app/src/main/java/org/cerion/stockcharts/ui/watchlist/RecyclerViewAdapter.kt
package org.cerion.stockcharts.ui.watchlist

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.cerion.stockcharts.databinding.ListItemWatchBinding

internal class RecyclerViewAdapter() : RecyclerView.Adapter<RecyclerViewAdapter.ListItemViewHolder>() {

    private var items = emptyList<WatchItemViewModel>()

    fun setItems(items: List<WatchItemViewModel>) {
        this.items = items
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListItemViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val binding = ListItemWatchBinding.inflate(layoutInflater, parent, false)
        return ListItemViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ListItemViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    inner class ListItemViewHolder(var binding: ListItemWatchBinding) : RecyclerView.ViewHolder(binding.root), View.OnClickListener {

        init {
            //binding.weekRange.isEnabled = false
            //binding.yearRange.isEnabled = false
            binding.root.setOnClickListener(this)
        }

        fun bind(viewModel: WatchItemViewModel) {
            //binding.viewModel = viewModel
            //binding.executePendingBindings()
        }

        override fun onClick(v: View) { // TODO open charts activity/fragment
            //Intent intent = ChartsActivity.newIntent(v.getContext(), binding.getViewModel().getSymbol(), binding.getViewModel().getChart());
            //v.getContext().startActivity(intent);
        }
    }
}